src/foam/core/types.js

```diff
@@ -490,7 +490,13 @@ foam.CLASS({
 foam.CLASS({
   package: 'foam.core',
   name: 'UnitValue',
-  extends: 'Long'
+  extends: 'Long',
+  properties: [
+    {
+      class: 'String',
+      name: 'unitPropName'
```

Comment:
how does unitPropName work?

Reply:
The `tableCellFormatter` will use it find the property that has the Currency String
An example is here in `Invoice.js`: https://github.com/nanoPayinc/NANOPAY/pull/8189/files

Reply:
@jlhughes This is the idea that Kate and I came up with. The problem is that we still need to store the exact unit (eg: CAD vs USD) as data on each instance of an object with a `UnitValue` property. So we would either need to make `UnitValue` store both the long (quantity) and the string (unit id) as its value, or continue what we're currently doing and require a second property that accompanies the `UnitValue` property and that stores the unit id. Eg: `amount` and `denomination`. Then all we're missing is a way to relate the two properties, which `unitPropName` accomplishes.

It's not the only potential solution to the problem, but it's very close to what we have now so very few changes are required, and none of them should be breaking. And it still allows us to define the `tableCellFormatter` once in a generic way and avoid repeating ourselves.

Reply:
can the default 'unitPropName' be 'denomination' ?

Reply:
That seems fine to me.
@KateMengjunXia could you please add a documentation comment to this property to explain what it's for?